drivers/usb/usbxhci/usbxhci.c

```diff
@@ -0,0 +1,572 @@
+    _In_ PVOID XhciTransfer,
+    _In_ PUSBPORT_SCATTER_GATHER_LIST SgList)
+{
+    DPRINT("XHCI_SubmitTransfer: UNIMPLEMENTED. FIXME\n");
+    return MP_STATUS_SUCCESS;
+}
+
+MPSTATUS
+NTAPI
+XHCI_SubmitIsoTransfer(
+    _In_ PVOID XhciExtension,
+    _In_ PVOID XhciEndpoint,
+    _In_ PUSBPORT_TRANSFER_PARAMETERS TransferParameters,
+    _In_ PVOID XhciTransfer,
+    _In_ PVOID isoParameters)
+{
+    DPRINT("XHCI_SubmitIsoTransfer: UNIMPLEMENTED. FIXME\n");
+    return MP_STATUS_SUCCESS;
+}
+
+VOID
+NTAPI
+XHCI_AbortTransfer(
+    _In_ PVOID XhciExtension,
+    _In_ PVOID XhciEndpoint,
+    _In_ PVOID XhciTransfer,
+    _In_ PULONG CompletedLength)
+{
+    DPRINT("XHCI_AbortTransfer: UNIMPLEMENTED. FIXME\n");
+}
+
+ULONG
+NTAPI
+XHCI_GetEndpointState(
+    _In_ PVOID XhciExtension,
+    _In_ PVOID XhciEndpoint)
+{
+    DPRINT("XHCI_GetEndpointState: UNIMPLEMENTED. FIXME\n");
+    return 0;
+}
+
+VOID
+NTAPI
+XHCI_SetEndpointState(
+    _In_ PVOID XhciExtension,
+    _In_ PVOID XhciEndpoint,
+    _In_ ULONG EndpointState)
+{
+    DPRINT("XHCI_SetEndpointState: UNIMPLEMENTED. FIXME\n");
+}
+
+VOID
+NTAPI
+XHCI_PollEndpoint(
+    _In_ PVOID XhciExtension,
+    _In_ PVOID XhciEndpoint)
+{
+    DPRINT("XHCI_PollEndpoint: UNIMPLEMENTED. FIXME\n");
+}
+
+VOID
+NTAPI
+XHCI_CheckController(_In_ PVOID XhciExtension)
+{
+    DPRINT("XHCI_CheckController: UNIMPLEMENTED. FIXME\n");
+}
+
+ULONG
+NTAPI
+XHCI_Get32BitFrameNumber(_In_ PVOID XhciExtension)
+{
+    DPRINT("XHCI_Get32BitFrameNumber: UNIMPLEMENTED. FIXME\n");
+    return 0;
+}
+
+VOID
+NTAPI
+XHCI_InterruptNextSOF(_In_ PVOID XhciExtension)
+{
+    DPRINT("XHCI_InterruptNextSOF: UNIMPLEMENTED. FIXME\n");
+}
+
+VOID
+NTAPI
+XHCI_EnableInterrupts(_In_ PVOID XhciExtension)
+{
+    DPRINT("XHCI_EnableInterrupts: UNIMPLEMENTED. FIXME\n");
+}
+
+VOID
+NTAPI
+XHCI_DisableInterrupts(_In_ PVOID XhciExtension)
+{
+    DPRINT("XHCI_DisableInterrupts: UNIMPLEMENTED. FIXME\n");
+}
+
+VOID
+NTAPI
+XHCI_PollController(_In_ PVOID XhciExtension)
+{
+    DPRINT("XHCI_PollController: UNIMPLEMENTED. FIXME\n");
+}
+
+VOID
+NTAPI
+XHCI_SetEndpointDataToggle(
+    _In_ PVOID XhciExtension,
+    _In_ PVOID XhciEndpoint,
+    _In_ ULONG DataToggle)
+{
+    DPRINT("XHCI_SetEndpointDataToggle: UNIMPLEMENTED. FIXME\n");
+}
+
+ULONG
+NTAPI
+XHCI_GetEndpointStatus(
+    _In_ PVOID XhciExtension,
+    _In_ PVOID XhciEndpoint)
+{
+    DPRINT("XHCI_GetEndpointStatus: UNIMPLEMENTED. FIXME\n");
+    return 0;
+}
+
+VOID
+NTAPI
+XHCI_SetEndpointStatus(
+    _In_ PVOID XhciExtension,
+    _In_ PVOID XhciEndpoint,
+    _In_ ULONG EndpointStatus)
+{
+    DPRINT("XHCI_SetEndpointStatus: UNIMPLEMENTED. FIXME\n");
+}
+
+NTSTATUS
+NTAPI
+DriverEntry(
+    _In_ PDRIVER_OBJECT DriverObject,
+    _In_ PUNICODE_STRING RegistryPath)
+{
+    DPRINT("DriverEntry: DriverObject - %p, RegistryPath - %wZ\n", DriverObject, RegistryPath);
+
+    RtlZeroMemory(&RegPacket, sizeof(USBPORT_REGISTRATION_PACKET));
```

Comment:
Is this RegPacket is a global variable?

```suggestion
    RtlZeroMemory(&RegPacket, sizeof(RegPacket));
```

Reply:
(Question still valid -- It's because I've the feeling it's not a good thing.)

Reply:
It's global, and it needs to be because it contains callback function pointers that usbport fills in.